### PR TITLE
chore(litmus-portal): Editor update with time-stamp based workflow mapping.

### LINKED ISSUE #1886: (litmus-portal): Refractor and update Editor as per new designs and add UNIX time-stamp to name as replacement of generateName.
- Refractor and update Editor as per new designs for the portal.
- Replace generateName with definitive name.
- Add UNIX time-stamp at the end of name on editor load.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
## Proposed changes

Refractor and update Editor as per new designs and add UNIX time-stamp to name as replacement of generateName.

fixes #1886 

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- None.

## Special notes for your reviewer:
High priority PR for workflow mapping. Needed before RC1.

### CODE REVIEW

**Comment by ishangupta-ds** [comment]
@rajdas98 @arkajyotiMukherjee @S-ayanide @amityt @gdsoumya @Jonsy13 @oumkale  PTAL.

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 1]
```diff
@@ -3,35 +3,57 @@ import { makeStyles, Theme } from '@material-ui/core';
...
-    backgroundColor: theme.palette.common.black,
+    backgroundColor: theme.palette.editorBackground,
     color: theme.palette.secondary.contrastText,
     width: '100%',
   },
 
   horizontalLineWhite: {
     marginTop: theme.spacing(4),
-    backgroundColor: theme.palette.secondary.contrastText,
+    backgroundColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:45]: we have a customColors in theme and have a white function there where you can pass the opacity

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 2]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+      marginTop: theme.spacing(1.5),
+    },
+  },
+
+  editorButtonFind: {
+    borderRadius: 3,
+    backgroundColor: theme.palette.editorBackground,
+    boxSizing: 'border-box',
+    color: theme.palette.secondary.contrastText,
+    borderColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:150]: same as above

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 3]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+    },
+  },
+
+  editorButtonFold: {
+    borderRadius: 3,
+    backgroundColor: theme.palette.editorBackground,
     boxSizing: 'border-box',
     color: theme.palette.secondary.contrastText,
-    borderColor: theme.palette.secondary.contrastText,
+    borderColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:169]: same as above

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 4]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+      marginTop: theme.spacing(1.5),
+    },
+  },
+
+  editorButtonUnfold: {
+    borderRadius: 3,
+    backgroundColor: theme.palette.editorBackground,
+    boxSizing: 'border-box',
+    color: theme.palette.secondary.contrastText,
+    borderColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:186]: same

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 5]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
   },
 
   editorButtonUndo: {
     borderRadius: 3,
-    backgroundColor: '#161616',
+    backgroundColor: theme.palette.editorBackground,
     boxSizing: 'border-box',
     color: theme.palette.secondary.contrastText,
-    borderColor: theme.palette.secondary.contrastText,
+    borderColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:203]: same

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 6]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
   },
 
   editorButtonDownload: {
     borderRadius: 3,
-    backgroundColor: '#161616',
+    backgroundColor: theme.palette.editorBackground,
     boxSizing: 'border-box',
     color: theme.palette.secondary.contrastText,
-    borderColor: theme.palette.secondary.contrastText,
+    borderColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:221]: same

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 7]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
   },
 
   editorButtonReplace: {
     borderRadius: 3,
-    backgroundColor: '#161616',
+    backgroundColor: theme.palette.editorBackground,
     boxSizing: 'border-box',
     color: theme.palette.secondary.contrastText,
-    borderColor: theme.palette.secondary.contrastText,
+    borderColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:239]: same

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 8]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
   },
 
   editorButtonSelectAll: {
     borderRadius: 3,
-    backgroundColor: '#161616',
+    backgroundColor: theme.palette.editorBackground,
     boxSizing: 'border-box',
     color: theme.palette.secondary.contrastText,
-    borderColor: theme.palette.secondary.contrastText,
+    borderColor: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:264]: same

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 9]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
   },
 
   editorButtonFullScreen: {
     borderRadius: 3,
-    backgroundColor: '#161616',
+    backgroundColor: theme.palette.editorBackground,
     boxSizing: 'border-box',
-    color: theme.palette.secondary.contrastText,
-    borderColor: '#161616',
+    color: 'rgba(255, 255, 255, 0.2)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:291]: same

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 10]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
     paddingLeft: theme.spacing(3.75),
     paddingBottom: theme.spacing(1.875),
+    marginTop: theme.spacing(-2),
     width: '1.875rem',
     height: '1.875rem',
-    marginLeft: theme.spacing(1.25),
+    marginLeft: theme.spacing(-7.5),
+    [theme.breakpoints.down('sm')]: {
+      marginLeft: theme.spacing(-2),
+      width: '0rem',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:301]: 0 rem is not needed replace with width: 0,
```suggestion
      width: 0,
```

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 11]
```diff
@@ -40,87 +62,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+      height: '0rem',
+      padding: theme.spacing(0),
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:303]: theme.spacing(0) is not needed just use 0
same goes for magins...do it everywhere
```suggestion
      padding: 0,
```

File: litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts [hunk 12]
```diff
@@ -6,9 +6,7 @@ const useStyles = makeStyles((theme: Theme) => ({
     width: '70%',
     padding: '1rem',
     margin: '2rem auto',
-    background: props.isDark
-      ? theme.palette.common.black
-      : theme.palette.common.white,
+    background: props.isDark ? '#1C1C1C' : theme.palette.common.white,
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts:9]: use color from theme data....editorBackground

File: litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts [hunk 13]
```diff
@@ -30,11 +28,13 @@ const useStyles = makeStyles((theme: Theme) => ({
     minHeight: 0,
     minWidth: 0,
     borderRadius: 3,
-    color: props.isDark ? 'rgba(255, 255, 255, 0.4)' : 'rgba(0, 0, 0, 0.4)',
+    color: props.isDark
+      ? theme.palette.secondary.contrastText
+      : 'rgba(0, 0, 0, 0.4)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts:33]: same

File: litmus-portal/frontend/src/theme/index.tsx [hunk 14]
```diff
@@ -103,6 +105,7 @@ function customTheme(options: ThemeOptions) {
           active: 'rgba(16, 155, 103, 0.1)',
         },
       },
+      editorBackground: '#1C1C1C',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/theme/index.tsx:108]: if this is already in theme...why is this not used?

File: litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts (hunk 12 above)
Comment (oumkale) [inline litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts:9]: @ishangupta-ds will you see the corresponding theme color for #1C1C1C.

File: litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts [hunk 15]
```diff
@@ -30,11 +28,13 @@ const useStyles = makeStyles((theme: Theme) => ({
...
     border: props.isDark
-      ? '1px solid rgba(255, 255, 255, 0.4)'
+      ? '1px solid rgba(255, 255, 255, 0.2)'
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Unimodal/styles.ts:35]: same

**Review by arkajyotiMukherjee — DISMISSED** [review]

**Comment by oumkale** [comment]
I observe that 'rgba(255, 255, 255, 0.2)' is used more than 10 times please will use the theme for borderColor.

File: litmus-portal/frontend/src/theme/index.tsx (hunk 14 above)
Comment (oumkale) [inline litmus-portal/frontend/src/theme/index.tsx:108]: Same

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 16]
```diff
@@ -3,35 +3,57 @@ import { makeStyles, Theme } from '@material-ui/core';
...
 
   statusDescription: {
-    width: '100%',
+    width: '95%',
     marginTop: theme.spacing(1.875),
-    fontFamily: 'Ubuntu',
     fontSize: '0.875rem',
     marginLeft: theme.spacing(2),
+    marginRight: theme.spacing(2),
+    fontStyle: 'normal',
```
Comment (oumkale) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:32]: By default it uses fontStyle is normal. You may remove this

**Comment by ishangupta-ds** [comment]
Changes have been made PTAL @arkajyotiMukherjee @oumkale @rajdas98

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 17]
```diff
@@ -40,87 +58,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
     overflow: 'auto',
-    maxHeight: '34.5rem',
-    width: '92%',
+    height: '50vh',
+    width: '100%',
+    '&::-webkit-scrollbar': {
+      width: '0.2em',
+    },
+    '&::-webkit-scrollbar-track': {
+      webkitBoxShadow: 'inset 0 0 6px rgba(0,0,0,0.00)',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:67]: why an rgba of 0.00???
use theme color here...if it doesn't need an opacity use `common.black` color

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 18]
```diff
@@ -40,87 +58,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+    },
+    '&::-webkit-scrollbar-thumb': {
+      backgroundColor: theme.palette.secondary.dark,
+      outline: '1px solid slategrey',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:71]: don't use colors that are not in theme...like slategrey
if this particular color is in design add it to theme properties or use a different shade of grey... mui default theme has a bunch of grey shades

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 19]
```diff
@@ -40,87 +58,268 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+      marginLeft: 0,
+      marginTop: theme.spacing(1.5),
+    },
+  },
+
+  editorButtonCopy: {
+    borderRadius: 3,
+    backgroundColor: theme.palette.editorBackground,
+    boxSizing: 'border-box',
+    color: theme.palette.secondary.contrastText,
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:129]: I can see a lot of common styles between the editor buttons...
this common css properties can be extracted into a single class and you can apply multiple class names in the HTML part using javascript template literals
like this
```
className={`${editorButton} ${editorButtonCopy}`}
```

File: litmus-portal/frontend/src/components/YamlEditor/styles.ts [hunk 20]
```diff
@@ -129,13 +328,25 @@ const useStyles = makeStyles((theme: Theme) => ({
...
-  markStyle: {
+  markStyleWrong: {
     display: 'inline-block',
     fontFamily: 'Ubuntu',
     fontSize: '1rem',
+    color: theme.palette.error.dark,
+  },
+
+  fullScreenGrid: {
+    width: '0%',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/YamlEditor/styles.ts:349]: change 0% to 0

**Review by arkajyotiMukherjee — DISMISSED** [review]

**Comment by ishangupta-ds** [comment]
@arkajyotiMukherjee PTAL. cc: @rajdas98

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by S-ayanide — APPROVED** [review]

**Review by oumkale — APPROVED** [review]

**Pull request merged**